backend/graphql/resolvers/courseResolvers.ts

```diff
@@ -43,11 +43,22 @@ const courseResolvers = {
     course: async (
       _parent: undefined,
       { id }: { id: string },
+      context: ExpressContext,
     ): Promise<CourseResponseDTO> => {
-      return courseService.getCourse(id);
+      const accessToken = getAccessToken(context.req);
+      if (!accessToken) {
+        throw new AuthenticationError(
+          "Failed authentication and/or authorization by role",
+        );
+      }
+
+      const role = await authService.getUserRoleByAccessToken(accessToken);
+      const attributes = getCourseVisibilityAttributes(role);
```

Comment:
Since courses will no longer have a published attribute (that's only for courses), this is a bit out of date no?